Dragonflight/ShamanElemental.lua

```diff
@@ -711,7 +755,21 @@ spec:RegisterHook( "reset_precast", function ()
             end
         end
     end
-
+	
+	-- Heat wave has a hidden aura, so we look at the last cast of primordial wave instead
+	if talent.heat_wave.enabled then
+		local applied = action.primordial_wave.lastCast
```

Comment:
Ah, looks like you handled my last comment already.  Great!

Reply:
Need to loop and cause each Heat Wave trigger, though.  This will only account for the first one.

Reply:
One thing this doesn't solve is that if the APL were to want to check ''buff.heat_wave.up" it would not function. I presume in the reset_precast we can set those correctly?

Reply:
I've updated the code to also set the ''buff.heat_wave" .applied, .up and .remains correctly.

One thing i've noticed, the recommendations still seem to be ''surprised'' whenever lava surge procs. It's not correctly registering that these procs are guaranteed to happen in 3/6/9/12 seconds. 

Not sure if that's something I'm personally doing wrong in this code, or if the tier28 implementation of this never really did that correctly anyway.